### PR TITLE
HTTP aco memory issues

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/eventer/eventer_epoll_impl.c [hunk 1]
```diff
@@ -448,8 +453,9 @@ static void eventer_epoll_impl_trigger(eventer_t e, int mask) {
       }
       master_fds[fd].e = NULL;
     }
-    eventer_free(e);
+    eventer_deref(e);
   }
+  eventer_deref(e);
```
Comment (bdunavant) [inline src/eventer/eventer_epoll_impl.c]: Won't the fall-through from line 456 cause this to deref twice?  This is in a few other places too.

File: src/examples/aco.c [hunk 2]
```diff
@@ -69,14 +69,18 @@ parse_cli_args(int argc, char * const *argv) {
...
 void subcall2(mtev_http_rest_closure_t *restc) {
   eventer_aco_gate_t gate = eventer_aco_gate();
   mtevL(mtev_error, "subcall2 entry\n");
-  eventer_aco_simple_asynch_gated(gate, asynch_hello, restc);
-  eventer_aco_simple_asynch_gated(gate, asynch_hello, restc);
+  int count = 2;
+  const char *count_str = mtev_http_request_querystring(mtev_http_session_request(restc->http_ctx), "count");
+  if(count_str) count = atoi(count_str);
+  for(int i=0; i<count; i++) {
+    eventer_aco_simple_asynch_gated(gate, asynch_hello, restc);
```
Comment (bdunavant) [inline src/examples/aco.c]: If for some reason count is invalid, is it bad if this runs 0 times?

Comment (postwait) [reply src/examples/aco.c]: Count=0 is one of my test cases.. It works as expected.

File: src/eventer/eventer_epoll_impl.c (hunk 1 above)
Comment (postwait) [reply src/eventer/eventer_epoll_impl.c]: Correct. Intentional... The first one is the deref b/c the event is no longer used.  The second one is the deref matching the ref we did in the beginning of the function.

**Pull request merged**